experiments/experiments.go

```diff
@@ -0,0 +1,323 @@
+	Owner string `json:"owner"`
+	// Enabled if set to false will disable the experiment and calls to Variant
+	// will always returns an empty string.
+	Enabled *bool `json:"enabled"`
+	// Version is the string to identify the specific version of the
+	// experiment.
+	Version string `json:"version"`
+	// Type specifies the type of experiment to run. If this value is not
+	// recognized, the experiment will be considered disabled.
+	Type string `json:"type"`
+	// StartTimestamp is a float of seconds since the epoch of date and time
+	// when you want the experiment to start. If an experiment has not been
+	// started yet, it is considered disabled.
+	StartTimestamp int64 `json:"start_ts"`
+	// StopTimestamp is a float of seconds since the epoch of date and time when
+	// you want the experiment to stop. Once an experiment is stopped, it is
+	// considered disabled.
+	StopTimestamp int64 `json:"stop_ts"`
+	// Experiment is the specific experiment.
+	Experiment ParsedExperiment `json:"experiment"`
+}
+
+// SimpleExperiment is a basic experiment choosing from a set of variants.
+type SimpleExperiment struct {
+	// id is the experiment identifier and should be unique.
+	id int
+	// name is a human-readable name of the experiment.
+	name string
+	// bucketSeed if provided, this provides the bucketSeed for determining which bucket a
+	// variant request lands in. Providing a consistent bucket bucketSeed will ensure
+	// a user is bucketed consistently. Calls to the variant method will return
+	// consistent results for any given bucketSeed.
+	bucketSeed string
+	// numBuckets determines how many available buckets there are for bucketing
+	// requests. This should match the numBuckets in the provided VariantSet.
+	// The default value is 1000, which provides a potential variant
+	// granularity of 0.1%.
+	numBuckets int
+	// enabled sets whether or not this experiment is enabled. disabling an
+	// experiment means all variant calls will return an empty string.
+	enabled bool
+	// startTime determines when this experiment is due to start. Variant
+	// requests prior to this time will return an empty string.
+	startTime time.Time
+	// endTime determines when this experiment is due to end. Variant requests
+	// after this time will return an empty string.
+	endTime time.Time
+	// logBucketing determines whether bucketing events should be logged.
+	logBucketing bool
+	// bucketVal is a string used for shifting the deterministic bucketing
+	// algorithm.  In most cases, this will be an Account's fullname.
+	bucketVal string
+	// variantSet contains a set of experimental variants as well as their
+	// distributions. It is used by experiments to track which bucket a variant
+	// is assigned to.
+	variantSet VariantSet
+	// targeting allows to target users with multiple parameters supporting
+	// both AND and OR based logical grouping.
+	targeting Targeting
+	// overrides if matched allow to force a particular variant.
+	overrides []map[string]Targeting
```

Comment:
It would be nice to get the documentation. I was considering exporting those fields just for that reason but from the specification it might be better to keep those private.